### PR TITLE
Allow zoneinfo objects

### LINKED ISSUE #915: Support zoneinfo.ZoneInfo timezones
In this block, it appears there's currently no handling for the standard library zoneinfo objects

https://github.com/dask/fastparquet/blob/fb545a5d8147eb111eded0d5ac11eda03c574134/fastparquet/util.py#L422-L435

Would be great to include a 

```python
 elif isinstance(dtype.tz, zoneinfo.ZoneInfo): 
     extra_metadata = {'timezone': dtype.tz.key} 
```

### PR DESCRIPTION
**Pull request by mroeschke**
closes #915

### CODE REVIEW

File: fastparquet/util.py [hunk 1]
```diff
@@ -417,25 +418,28 @@ def get_column_metadata(column, name, object_dtype=None):
...
-                extra_metadata = {'timezone': stz}
-            elif "Offset" in stz:
-                extra_metadata = {'timezone': f"{dtype.tz._minutes // 60:+03}:00"}
-            else:
-                raise KeyError
-        except Exception as e:
-            raise ValueError("Time-zone information could not be serialised: "
-                             "%s, please use another" % str(dtype.tz)) from e
+    elif isinstance(dtype, pd.DatetimeTZDtype):
+        if isinstance(dtype.tz, zoneinfo.ZoneInfo):
```
Comment (martindurant) [inline fastparquet/util.py:422]:
```suggestion
        if getattr(dtype.tz, "zone", False):
```
?

**Comment by martindurant** [comment]
update cibuildwheel to 2.16.5 ?

Aside from the wheels, a name is coming out as type int in writer, which is not allowed by parquet (in test_tz_zoneinfo)

**Comment by martindurant** [comment]
There is some dask config option that says "don't use dask-expr" we'll need to set, since they didn't bother to implement fastparquet stuff (that already existed).

**Comment by martindurant** [comment]
Still hitting dask-expr code somehow:
```
../../../micromamba-root/envs/test_env/lib/python3.10/site-packages/dask_expr/_collection.py:3152: in to_parquet
    return to_parquet(self, path, **kwargs)
../../../micromamba-root/envs/test_env/lib/python3.10/site-packages/dask_expr/io/parquet.py:335: in to_parquet
    engine = _set_parquet_engine(engine=engine, meta=df._meta)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

engine = 'fastparquet', meta = Empty DataFrame
Columns: [unique, id]
Index: []

    def _set_parquet_engine(engine=None, meta=None):
        # Use `engine` or `meta` input to set the parquet engine
        if engine == "fastparquet":
>           raise NotImplementedError("Fastparquet engine is not supported")
E           NotImplementedError: Fastparquet engine is not supported
```

**Comment by mroeschke** [comment]
Added an `xfail` on `test_sorted_row_group_columns_with_filters` for now since it doesn't seem directly related to the `zoneinfo` support directly

**Comment by martindurant** [comment]
Would you also like to fix the
> ValueError: assignment destination is read-only

errors from new pandas?

**Comment by mroeschke** [comment]
I would prefer to address that in a follow up to avoid to scope creep in this PR

**Pull request merged**